libclamav/clamav.h

```diff
@@ -601,6 +601,63 @@ enum cl_msg {
     CL_MSG_ERROR        = 128 /* LibClamAV ERROR: */
 };
 
+/**
+ * @brief Progress callback for sig-load, engine-compile, and engine-free.
+ *
+ * Disclaimer: total_items may increase with each call as new items to be
+ * processed are found. As such, total_items == now_completed does not mean it
+ * guaranteed to be complete.
```

Comment:
I should remove this disclaimer. I found away around this concern to pre-calculate all the compile tasks and all the free tasks. I also can add 1 task at the beginning and remove 1 task at the end if necessary to ensure that `total_items == now_completed` means we really are done... so yeah this disclaimer is now a lie, and needs to be removed.

Reply:
@craig65535 Are you okay with the progress callback not returning anything?  Would you want to be able to return something to cancel the task (or allow it to continue)?

Reply:
@micahsnyder As discussed, I think it would be good to future-proof this and have the callback return a `cl_error_t` (which would be ignored for now).

Reply:
Ok just added a commit that adds a reserved return for the callback.